data.go

```diff
@@ -40,12 +41,7 @@ func (td *TrianglesData) SetLen(len int) {
 	if len > td.Len() {
 		needAppend := len - td.Len()
 		for i := 0; i < needAppend; i++ {
-			*td = append(*td, struct {
-				Position  Vec
-				Color     RGBA
-				Picture   Vec
-				Intensity float64
-			}{ZV, Alpha(1), ZV, 0})
+			*td = append(*td, vertex{})
```

Comment:
Do you mean the unnamed struct type, or the literal with actual contents? I don't like unnamed struct types which are used more than once, because that creates a possible future maintenance headache.